test/bordeaux-threads-test.lisp

```diff
@@ -273,6 +273,139 @@ Distributed under the MIT license (see LICENSE file)
+    (is (equalp  4 (bt:receive-message mailbox)))
+    (is (equalp  "zielone" (bt:receive-message-no-hang mailbox)))
+    (is (= 1 (bt:mailbox-count mailbox)))
+    (is (equalp "koty" (bt:receive-message-no-hang mailbox)))
+    ;; Ensure empty box behavior
+    (is (= 0 (bt:mailbox-count mailbox)))
+    (is (bt:mailbox-empty-p mailbox))
+    (is (null (receive-message-no-hang mailbox)))
+    (is (null (receive-message mailbox :timeout 0.1)))
+    (is (= 0 (bt:mailbox-count mailbox)))
+    (is (bt:mailbox-empty-p mailbox))
+    ;; add some new data and verify group operations
+    (dotimes (v 10) (send-message mailbox v))
+    (is (= 10 (bt:mailbox-count mailbox)))
+    (is (not (bt:mailbox-empty-p mailbox)))
+    (is (equalp '(0 1 2 3 4 5 6 7 8 9) (list-mailbox-messages mailbox)))
+    (is (null (receive-pending-messages mailbox 0)))
+    (signals type-error (receive-pending-messages mailbox -1))
+    (is (= 10 (bt:mailbox-count mailbox)))
+    (is (not (bt:mailbox-empty-p mailbox)))
+    (is (equalp '(0 1 2 3 4) (receive-pending-messages mailbox 5)))
+    (is (equalp '(5 6 7 8 9) (list-mailbox-messages mailbox)))
+    (is (= (bt:mailbox-count mailbox) 5))
+    (is (not (bt:mailbox-empty-p mailbox)))
+    (is (equalp '(5 6 7 8 9) (receive-pending-messages mailbox 20)))
+    (is (= (bt:mailbox-count mailbox) 0))
+    (is (bt:mailbox-empty-p mailbox))
+    (is (null (receive-pending-messages mailbox 20)))
+    (is (= 0 (bt:mailbox-count mailbox)))
+    (is (bt:mailbox-empty-p mailbox))
+    ;; receive-pending-messages without optional argument
+    (dotimes (v 10) (send-message mailbox v))
+    (is (= 10 (bt:mailbox-count mailbox)))
+    (is (not (bt:mailbox-empty-p mailbox)))
+    (is (equalp '(0 1 2 3 4 5 6 7 8 9) (receive-pending-messages mailbox)))
+    (is (= 0 (bt:mailbox-count mailbox)))
+    (is (bt:mailbox-empty-p mailbox))))
+
+(test mailbox-producer-consumers
+  (let (;; we start from 0 and always send messages
+        ;; monotonically increasing. That means that each
+        ;; consumer has monotonically increasing
+        ;; elements. Note that results-mailbox may have all
+        ;; that jumbled up, because different consumers may
+        ;; send at different times.
+        (bt:mailbox (make-mailbox :initial-contents '(0)))
+        ;; we will submit all results to the second mailbox,
+        ;; then sort them and verify if nothing got lost. This
+        ;; also gives us concurrent send-message.
+        (results-mailbox (make-mailbox))
+        ;; all errors in a thread should be send here. For
+        ;; intance if messages are not monotonous from the
+        ;; consumer perspective it is an ordering issue.
+        (errors-mailbox (make-mailbox))
+        ;; flag to tell consumers that we are full. We always
+        ;; have a timeout in make-1-consumer (and
+        ;; make-n-consumer returns immedietely), so we may
+        ;; safely assume that join-thread will return.
+        (konsument-je-żeby-jeść-p nil)
```

Comment:
> `(konsument-je-żeby-jeść-p nil)`

Nie sądzę, żeby nazwa tej zmiennej była zrozumiała dla osób niepolskojęzycznych.

Reply:
it doesn't matter, it is clearly the variable indicating that processing has ended.

Reply:
I'd still rename it to English, even if just to avoid causing trouble for people who don't have Polish ogonkis on their keynaps. Symbol names in international code should be ASCII enough to be typable in the keyboard without copy-pasting.